scrapy/downloadermiddlewares/redirect.py

```diff
@@ -55,28 +56,26 @@ class RedirectMiddleware(BaseRedirectMiddleware):
 
     def process_response(self, request, response, spider):
         if (request.meta.get('dont_redirect', False) or
-               response.status in getattr(spider, 'handle_httpstatus_list', []) or
-               response.status in request.meta.get('handle_httpstatus_list', []) or
-               request.meta.get('handle_httpstatus_all', False)):
+                response.status in getattr(spider, 'handle_httpstatus_list', []) or
+                response.status in request.meta.get('handle_httpstatus_list', []) or
+                request.meta.get('handle_httpstatus_all', False)):
             return response
 
-        if request.method == 'HEAD':
-            if response.status in [301, 302, 303, 307] and 'Location' in response.headers:
-                redirected_url = urljoin(request.url, response.headers['location'])
+        location = None
+        if 'Location' in response.headers:
+            # HTTP header is ascii or latin1, redirected url will be percent-encoded utf-8
+            location = to_native_str(response.headers['location'].decode('latin1'))
+
+        if location is not None and response.status in [301, 302, 303, 307]:
```

Comment:
`location is not None` check is not needed: it is not None iff `'Location' in response.headers` is True.

I'd also prefer to exit from function earlier if 'Location' in not in response.headers - there is no need for an extra level of indentation and `location = None` statement.